Source/BugsnagSessionTracker.h

```diff
@@ -17,25 +17,48 @@ typedef void (^SessionTrackerCallback)(BugsnagSession *newSession);
 
 @interface BugsnagSessionTracker : NSObject
 
+/**
+ Create a new session tracker
+
+ @param config The Bugsnag configuration to use
+ @param callback A callback invoked each time a new session is started
+ @return A new session tracker
+ */
 - (instancetype)initWithConfig:(BugsnagConfiguration *)config
-                     apiClient:(BugsnagSessionTrackingApiClient *)apiClient
-                      callback:(void(^)(BugsnagSession *))callback;
+            postRecordCallback:(void(^)(BugsnagSession *))callback;
 
-- (void)startNewSession:(NSDate *)date
-               withUser:(BugsnagUser *)user
-           autoCaptured:(BOOL)autoCaptured;
+/**
+ Record and send a new session
+ */
+- (void)startNewSession;
+
+/**
+ Record a new auto-captured session if neededed. Auto-captured sessions are only
+ recorded and sent if -[BugsnagConfiguration shouldAutoCaptureSessions] is YES
+ */
+- (void)startNewSessionIfAutoCaptureEnabled;
 
-- (void)suspendCurrentSession:(NSDate *)date;
-- (void)incrementHandledError;
-- (void)send;
-- (void)onAutoCaptureEnabled;
+/**
+ Handle the app foregrounding event. If more than 30s has elapsed since being
+ sent to the background, records a new session if session auto-capture is
+ enabled
+ */
+- (void)handleAppForegroundEvent;
 
-@property (readonly) BugsnagSession *currentSession;
-@property (readonly) BOOL isInForeground;
+/**
+ Handle the app backgrounding event. Tracks time between foreground and
+ background to determine when to automatically record a session.
+ */
+- (void)handleAppBackgroundEvent;
 
 /**
- * Called when a session is altered
+ Handle some variation of Bugsnag.notify() being called.
+ Increases the number of handled errors recorded for the current session, if
+ a session exists.
  */
-@property SessionTrackerCallback callback;
+- (void)handleHandledErrorEvent;
```

Comment:
Would it also make sense to have an unhandled equivalent of this method?

Reply:
Not really; if an unhandled event occurs, the app is crashing. There can’t be more than 1 unhandled event in a single session.